test/jdk/java/foreign/TestHandshake.java

```diff
@@ -98,16 +98,18 @@ static abstract class AbstractSegmentAccessor implements Runnable {
         @Override
         public final void run() {
             start("\"Accessor #\" + id");
-            outer: while (segment.scope().isAlive()) {
+            while (segment.scope().isAlive()) {
                 try {
                     doAccess();
                 } catch (IllegalStateException ex) {
-                    long delay = System.currentTimeMillis() - start.get();
-                    System.out.println("Accessor #" + id + " suspending - elapsed (ms): " + delay);
-                    backoff();
-                    delay = System.currentTimeMillis() - start.get();
-                    System.out.println("Accessor #" + id + " resuming - elapsed (ms): " + delay);
-                    continue outer;
+                    if (!failed.get()) {
+                        // ignore - this means segment was alive, but was closed while we were accessing it
+                        // next isAlive test should fail
```

Comment:
If we see the exception, we should be able to test that the scope is not alive here as well
```suggestion
                        // next isAlive test should fail
                        assertFalse(segment.scope().isAlive());
```